src/test/java/net/pms/dlna/DLNAMediaInfoTest.java

```diff
@@ -0,0 +1,152 @@
+package net.pms.dlna;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ch.qos.logback.classic.Level;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.Percentage.withPercentage;
+
+import net.pms.PMS;
+import net.pms.configuration.PmsConfiguration;
+import net.pms.configuration.RendererConfiguration;
+import net.pms.dlna.virtual.VirtualFolder;
+import net.pms.encoders.Player;
+import net.pms.encoders.PlayerFactory;
+import net.pms.logging.LoggingConfig;
+import net.pms.service.Services;
+
+
+public class DLNAMediaInfoTest
+{
+	private static final Class<?> CLASS = DLNAMediaInfoTest.class;
+	
+	private static final String[] test_files =
+	{
+			"pexels-video-4809.mp4", "pexels-video-4809.mkv"
+	};
+	private static final int[] test_content =
+			new int[test_files.length];
+
+	@BeforeClass
+	public static void SetUPClass()
+	{
+		PMS.configureJNA();
+		PMS.forceHeadless();
+		try {
+			PMS.setConfiguration(new PmsConfiguration(false));
+		} catch (Exception ex) {
+			throw new AssertionError(ex);
+		}
+		assert PMS.getConfiguration() != null;
+		PMS.getConfiguration().setSharedFolders(null);
+		PMS.getConfiguration().setScanSharedFoldersOnStartup(false);
+		PMS.getConfiguration().setUseCache(false);
+
+		LoggingConfig.setRootLevel(Level.TRACE);
+		// force unbuffered if in trace mode
+		LoggingConfig.setBuffered(false);
+		
+		Logger logger = LoggerFactory.getLogger(CLASS);
+
+		Services.create();
+		 try {
+			 PMS.getConfiguration().initCred();
+		} catch (Exception ex) {
+			logger.warn("Failed to write credentials configuration", ex);
+		}
+
+
+		if (PMS.getConfiguration().isRunSingleInstance()) {
+			PMS.killOld();
+		}
+
+		// Create the PMS instance returned by get()
+		PMS.get();
+		
+		// Create handles to the test content
+		// This comes from RequestV2::answer()
+
+		DLNAResource parent = new VirtualFolder("test","test");
+		parent.setDefaultRenderer(RendererConfiguration.getDefaultConf());
+		PMS.getGlobalRepo().add(parent);
```

Comment:
@stasinos I have the same problem like Travis. The GlobalRepo is not initialized and is `null`. Tested with Eclipse and WIN10

Reply:
Are you running the complete UMS test suite or this particular test only?
My current understanding of the issue is that if I run this test only, then all is well. But I do a complete mvn validate, then some other test has initialized PMS and my invokation of PMS.get() has no effect; but for some reason the global repo has disappeared in the meantime.

Reply:
@stasinos I am running all tests.

```
[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------------------< net.pms:ums >-----------------------------
[INFO] Building Universal Media Server 7.6.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce) @ ums ---
[INFO] 
[INFO] --- git-commit-id-plugin:2.2.3:revision (default) @ ums ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ ums ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 429 resources to resources
[INFO] Copying 1 resource to resources
[INFO] Copying 1 resource to ..
[INFO] 
[INFO] --- maven-antrun-plugin:1.8:run (process-resources-windows) @ ums ---
[INFO] Executing tasks

main:
[INFO] Executed tasks
[INFO] 
[INFO] --- maven-compiler-plugin:3.7.0:compile (default-compile) @ ums ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ ums ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource to resources
[INFO] Copying 15 resources to net/pms/util
[INFO] Copying 2 resources to net/pms/dlna
[INFO] 
[INFO] --- maven-compiler-plugin:3.7.0:testCompile (default-testCompile) @ ums ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 40 source files to E:\Workspace\UniversalMediaServer\target\test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.20:test (default-test) @ ums ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running net.pms.configuration.FormatConfigurationRegressionTest
[INFO] Tests run: 107, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.143 s - in net.pms.configuration.FormatConfigurationRegressionTest
[INFO] Running net.pms.configuration.PmsConfigurationTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.65 s - in net.pms.configuration.PmsConfigurationTest
[INFO] Running net.pms.configuration.RendererConfigurationTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.872 s - in net.pms.configuration.RendererConfigurationTest
[INFO] Running net.pms.database.TableFilesStatusTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.41 s - in net.pms.database.TableFilesStatusTest
[INFO] Running net.pms.dlna.DLNAMediaAudioTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in net.pms.dlna.DLNAMediaAudioTest
[INFO] Running net.pms.dlna.DLNAMediaInfoTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.035 s <<< FAILURE! - in net.pms.dlna.DLNAMediaInfoTest
[ERROR] net.pms.dlna.DLNAMediaInfoTest  Time elapsed: 0.035 s  <<< ERROR!
java.lang.NullPointerException
	at net.pms.dlna.DLNAMediaInfoTest.SetUPClass(DLNAMediaInfoTest.java:76)

[INFO] Running net.pms.dlna.DLNAMediaSubtitleTest
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.166 s - in net.pms.dlna.DLNAMediaSubtitleTest
[INFO] Running net.pms.dlna.LibMediaInfoParserTest
[INFO] Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.016 s - in net.pms.dlna.LibMediaInfoParserTest
[INFO] Running net.pms.formats.FormatsExtensionTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.031 s - in net.pms.formats.FormatsExtensionTest
[INFO] Running net.pms.formats.FormatsRegressionTest
[INFO] Tests run: 62, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.027 s - in net.pms.formats.FormatsRegressionTest
[INFO] Running net.pms.formats.v2.AudioAttributeTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in net.pms.formats.v2.AudioAttributeTest
[INFO] Running net.pms.formats.v2.AudioPropertiesTest
[INFO] Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 s - in net.pms.formats.v2.AudioPropertiesTest
[INFO] Running net.pms.formats.v2.SubtitleTypeTest
[INFO] Tests run: 17, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.027 s - in net.pms.formats.v2.SubtitleTypeTest
[INFO] Running net.pms.formats.v2.SubtitleUtilsTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.032 s - in net.pms.formats.v2.SubtitleUtilsTest
[INFO] Running net.pms.logging.LoggingTest
DEBUG 18:22:32.438 [main] LogBack started with configuration file: E:\Workspace\UniversalMediaServer\src\main\external-resources\logback.headless.xml
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.134 s - in net.pms.logging.LoggingTest
[INFO] Running net.pms.newgui.components.SpinnerIntModelTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in net.pms.newgui.components.SpinnerIntModelTest
[INFO] Running net.pms.newgui.components.TextAreaFIFOTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.219 s - in net.pms.newgui.components.TextAreaFIFOTest
[INFO] Running net.pms.newgui.LanguageSelectionTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.02 s - in net.pms.newgui.LanguageSelectionTest
[INFO] Running net.pms.newgui.ViewLevelTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in net.pms.newgui.ViewLevelTest
[INFO] Running net.pms.test.formats.FormatFactoryTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.036 s - in net.pms.test.formats.FormatFactoryTest
[INFO] Running net.pms.test.formats.FormatRecognitionTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.222 s - in net.pms.test.formats.FormatRecognitionTest
[INFO] Running net.pms.test.formats.FormatTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in net.pms.test.formats.FormatTest
[INFO] Running net.pms.test.Iso639Test
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 s - in net.pms.test.Iso639Test
[INFO] Running net.pms.test.TaskRunnerTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.005 s - in net.pms.test.TaskRunnerTest
[INFO] Running net.pms.util.FilePermissionsTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.023 s - in net.pms.util.FilePermissionsTest
[INFO] Running net.pms.util.FileUtilTest
[INFO] Tests run: 33, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.165 s - in net.pms.util.FileUtilTest
[INFO] Running net.pms.util.IteratorsTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in net.pms.util.IteratorsTest
[INFO] Running net.pms.util.jna.JNATypesTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.017 s - in net.pms.util.jna.JNATypesTest
[INFO] Running net.pms.util.jna.macos.corefoundation.CoreFoundationTest
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.001 s - in net.pms.util.jna.macos.corefoundation.CoreFoundationTest
[INFO] Running net.pms.util.jna.macos.iokit.IOKitTest
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0 s - in net.pms.util.jna.macos.iokit.IOKitTest
[INFO] Running net.pms.util.jna.macos.kernreturn.KernReturnTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.006 s - in net.pms.util.jna.macos.kernreturn.KernReturnTest
[INFO] Running net.pms.util.LanguagesTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in net.pms.util.LanguagesTest
[INFO] Running net.pms.util.PlayerUtilTest
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.006 s - in net.pms.util.PlayerUtilTest
[INFO] Running net.pms.util.RationalTest
[INFO] Tests run: 30, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.041 s - in net.pms.util.RationalTest
[INFO] Running net.pms.util.StringUtilTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in net.pms.util.StringUtilTest
[INFO] Running net.pms.util.UMSUtilsTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.062 s - in net.pms.util.UMSUtilsTest
[INFO] Running net.pms.util.VersionTest
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 s - in net.pms.util.VersionTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   DLNAMediaInfoTest.SetUPClass:76 NullPointer
[INFO] 
[ERROR] Tests run: 368, Failures: 0, Errors: 1, Skipped: 2
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 21.241 s
[INFO] Finished at: 2018-11-17T18:22:34+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.20:test (default-test) on project ums: There are test failures.
[ERROR] 
[ERROR] Please refer to E:\Workspace\UniversalMediaServer\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```